### ISSUE TITLE
CQL: Usage of Like

### ISSUE THREAD

**Issue by constantinius** [opengeospatial/ogcapi-features#555]
The CQL JSON encoding of Like is quite misleading and also different from the possibilities of CQL Text

For CQL text the `isLikePredicate` is defined as such:
```
isLikePredicate =  scalarExpression "LIKE" patternExpression
                   ["WILDCARD" quote alpha quote]
                   ["SINGLECHAR" quote alpha quote]
                   ["ESCAPECHAR" quote alpha quote]
                   ["NOCASE" booleanLiteral] ;
```

So the pattern is explicit and is not an expression unlike its JSON equivalent:
```json
      "isLikePredicate": {
         "type": "object",
         "required": ["like"],
         "properties": {
            "like": { "$ref": "#/$defs/scalarOperands" },
            "wildcard": { "type": "string", "default": "%" },
            "singleChar": { "type": "string", "default": "." },
            "escapeChar": { "type": "string", "default": "\\" },
            "nocase": { "type": "boolean", "default": true }
         }
      },

// ...

      "scalarOperands": {
         "type": "array",
         "minItems": 2,
         "maxItems": 2,
         "items": {
            "$ref": "#/$defs/scalarExpression"
         }
      },
```

Here it is unclear what is the value and what is the pattern and the pattern is actually an expression, giving much more possible variations.

I would propose a schema such as this:
```json
      "isLikePredicate": {
         "type": "object",
         "required": ["value", "pattern"],
         "properties": {
            "value":  { "#/$defs/scalarExpression" },
            "pattern":  { "type": "string"},
            "wildcard": { "type": "string", "default": "%" },
            "singleChar": { "type": "string", "default": "." },
            "escapeChar": { "type": "string", "default": "\\" },
            "nocase": { "type": "boolean", "default": true }
         }
      },

**Comment by cportele** [comment]
See also #541 and #547.

**Comment by pvretano** [comment]
Resolved by PR #592.  Note the PR also simplifies the LIKE operator so that only value & pattern are left in the JSON encoding.

**Comment by pvretano** [comment]
Issue resolved by PR #592.  See also issues #541 and #567.
